### DISCUSSION TITLE
A 'ExampleConvection' is not a registered object.

### DISCUSSION THREAD

**Discussion by engrmessi** [idaholab/moose#23838] [Q&A Getting Started] [answered]
I tried to run the ex02 file with sbatch file , then i got this error. ex01 file ran successfully with sbatch. please how do i fix this?

**Comment by GiudGiud** [comment, marked as answer]
Hello

you’re probably not using the executable for example 2

the code (example convection object) is specific to each example, these objects are not used anywhere else

**Reply by engrmessi** [reply]
I used the file ex02.i . is this the file you mean?

**Reply by GiudGiud** [reply]
It's not just the input file, there's code for the ExampleConvection object in the moose/examples/ex02_kernel folder

**Reply by libinbabu1991** [reply]
Hello, 
I also got the same error:
A 'ExampleConvection' is not a registered object.
I edited the parameter name as instructed based on my app name cat1
registerMooseObject("cat1App", ExampleConvection);

Still getting the same error. Please advise how to run the example 2 correctly.

-Libin

**Reply by GiudGiud** [reply]
hello

once you registered, did you recompile?
and are you using the executable in the folder that you compiled in?

**Reply by libinbabu1991** [reply]
Hi,
I didn't quite get what you mean by register. May be this is the issue?
As a new user, I need to familiarize with the process. Let me explain what has been done so far.

I copied only the ex02, mug.e amd the kernal from the example folder to my problem folder.
I also went to example problem folder in the Moose folder and changed the name to src/base/cat1App.  (cat1 is my app name)

When I run this in Moose, I get the same error message:
A 'ExampleConvection' is not a registered object.

**Reply by GiudGiud** [reply]
register is the "register" call at the top of the source of ExampleConvection

Did you recompile after changing anything in the source code?

**Reply by libinbabu1991** [reply]
If you mean changing the ExampleApp to Moose app in the below register parameter, yes.
registerMooseObject("MooseApp", ExampleConvection);

Yes, I recompiled as well.

**Reply by GiudGiud** [reply]
can you paste the compilation log here please?
and then the command you run in the directory, where you hit the error

**Reply by libinbabu1991** [reply]
Hi, 
I created a new app called dog. I could not run the example 2 from this new folder initially because of the error that I have mentioned above.

 However it is working now after making some changes as below. Not sure if they are the right steps for using in the next examples given in the Moose package.

1. Copied the ex02 and mug.e from the moose\examples\ex02_kernel to dog folder
2. Copied the Example.c from the moose\examples\ex02_kernel\src\kernels to dog folder
3. Copied the Example.h moose\examples\ex02_kernel\include\kernels to dog\include
4. Edited the Example.c content from registerMooseObject("ExampleApp", ExampleConvection); to registerMooseObject("dogApp", ExampleConvection);

**Reply by mehdikamyabi** [reply]
> Hello
> 
> you’re probably not using the executable for example 2
> 
> the code (example convection object) is specific to each example, these objects are not used anywhere else

Hello
I faced with the same error. May you please tell me how I should make an executable version of this example?

**Reply by GiudGiud** [reply]
Hello

In the folder of the example you should have a Makefile. 
You can type "make -j 4 (replace by the number of cores on your machine) to build the executable for the example

**Reply by mehdikamyabi** [reply]
Thank you for your quick reply and apologies if my questions are too basic.
I noticed that there are two Makefiles: one in the main Examples directory, which contains all tutorials (from ex01, ex02_kernel, ex03, up to ex21_debugging), and another inside the ex02_kernel directory itself. which folder do you mean?
However, I tried running the commands you suggested (make -j4 and also make without specifying the number of cores) in both locations, but I encountered the same error in each case:

Makefile:18: /home/kamyabi/projects/examples/moose/Ex1/framework/build.mk: No such file or directory
make: *** No rule to make target '/home/kamyabi/projects/examples/moose/Ex1/framework/build.mk'.  Stop.

may you please help me with this?

**Reply by GiudGiud** [reply]
hello

it's not finding moose because you moved the files
You can do
```
export MOOSE_DIR=/home/kamyabi/projects/moose
```
if that's where moose is
